### ISSUE TITLE
[Linux] process_iter is faster without attrs than with

### ISSUE THREAD

**Issue by OvervCW** [giampaolo/psutil#2200]
## Summary

* OS: Linux
* Architecture: 64bit
* Psutil version: 5.9.4
* Python version: Python 3.11.0
* Type: performance

## Description

[The documentation](https://psutil.readthedocs.io/en/latest/index.html?highlight=process_iter#psutil.process_iter) recommends using the `attr` argument to have `process_iter` only retrieve the info you need rather than leaving it empty and retrieve everything.

I've tested the performance of two snippets of code:

### Snippet A

```python
[p.name() for p in psutil.process_iter()]
```

### Snippet B

```python
[p.name() for p in psutil.process_iter(['name'])]
```

On my system, snippet A is actually (slightly) faster than snippet B. Is this expected?

**Comment by giampaolo** [comment]
>  Is this expected?

Yes, because it does some [additional computation](https://github.com/giampaolo/psutil/blob/7f5e0c25e026ba954d6b5179ab132602b271703b/psutil/__init__.py#L497-L539) compared to directly calling the desired method. I expect the difference to be quite minimal though, probably not significant. 

In reality, what makes the biggest difference when using `process_iter`, is the fact that for every process we check if it [is_running](https://github.com/giampaolo/psutil/blob/7f5e0c25e026ba954d6b5179ab132602b271703b/psutil/__init__.py#L1445), to make sure the PID has not been reused. Internally this will call Process' `create_time()` for every process, which is O(n). 

Because `process_iter()` is supposed to be called in a loop, and because PID reuse is a rare event which should occur between longer period of times between `process_iter()` calls, this check should probably be removed or changed.

**Comment by OvervCW** [comment]
Thanks for your insight. The difference is indeed very minimal in this example, but appeared to be more significant in another scenario in an actual application. I don't have a simple reproducible version of that yet, though.
